### PR TITLE
Add gaussian priors

### PR DESCRIPTION
**Pull request by mblomqvist**
This PR adds the option to apply gaussian priors on fit parameters. It's easy to add other types of priors in priors.py in future updates. Set the gaussian prior in the correlation ini file with e.g.

[priors]
beta_hcd = 0.5 0.2 gaussian

Prints a warning if the same prior is set in multiple ini files. The way the code is set up, only the parameter names are saved in 'list of prior pars'. Do we want to change that?

### CODE REVIEW

**Comment by mblomqvist** [comment]
For Lya-Lya using pk_hcd:

without prior on beta_hcd
Combined
1572.69403202 1590 11 0.540024659478
LYA LYA
1572.69403202 1590
L0_hcd [ 14.54085426   0.47259595]
ap [ 1.02654815  0.03458294]
at [ 0.97931388  0.04489884]
beta_LYA [ 1.53307142  0.14578328]
beta_hcd [ 0.88870877  0.54844373]
bias_LYA [-0.20158922  0.00822264]
bias_SiII(1190) [-0.00154361  0.00028116]
bias_SiII(1193) [-0.00115782  0.00027069]
bias_SiII(1260) [-0.00058728  0.00034017]
bias_SiIII(1207) [-0.0037754   0.00030167]
bias_hcd [-0.02881166  0.00736378]

with prior on beta_hcd
Combined
1573.24120383 1590 11 0.536148211771
LYA LYA
1573.1403622 1590
L0_hcd [ 14.45792086   0.45904712]
ap [ 1.02677539  0.03457822]
at [ 0.97930897  0.0448658 ]
beta_LYA [ 1.61540889  0.11378404]
beta_hcd [ 0.56351114  0.17529456]
bias_LYA [-0.20641433  0.00553253]
bias_SiII(1190) [-0.00155624  0.00028168]
bias_SiII(1193) [-0.00116218  0.00027158]
bias_SiII(1260) [-0.00059496  0.00034155]
bias_SiIII(1207) [-0.00376108  0.00030196]
bias_hcd [-0.03359536  0.00408609]

**Comment by mblomqvist** [comment]
The travis test is complaining because the results file needs to be regenerated (I think). However, we first need to decide on what to save about the priors. Furthermore, I can't run the unittest, as reported #351

**Comment by londumas** [comment]
I think it would be nice to same the parameters of the prior: name and value.

**Comment by ngbusca** [comment]
@mblomqvist could you avoid writing the list of priors if there are no priors set? This should fix the travis tests.

**Comment by ngbusca** [comment]
@mblomqvist could you also check that the code works if you specify more than one prior?

**Comment by mblomqvist** [comment]
@ngbusca I changed the output such that the list or priors is only created if there are any priors (i.e. users have to check the output file if the list exists before attempting to read it). Running with two priors works swimmingly.

**Comment by mblomqvist** [comment]
The code now passes the tests, so it's ready to be merged from that point of view. If we want to save the prior's values too, I'd need to add that first.

**Comment by ngbusca** [comment]
@mblomqvist yes, it would be great to save the prior values as well so we don't lose that information.

**Comment by mblomqvist** [comment]
@ngbusca I'm nearly done adding it.

**Comment by londumas** [comment]
@mblomqvist, just a simple comment if we change to:
```
[priors]
beta_hcd = gaussian 0.5 0.2
```
we can generelize easily to all type of priors with:
```
priors.prior_dic[item] = partial(getattr(priors, value[0]), prior_pars=sp.array(value[1:]).astype(float), name=item)
```

File: py/picca/fitter2/chi2.py [hunk 1]
```diff
@@ -228,6 +228,10 @@ def export(self):
         for (p1, p2), cov in self.best_fit.covariance.items():
             g.attrs["cov[{}, {}]".format(p1,p2)] = cov
 
+        if len(priors.prior_dic) != 0:
+            for prior in priors.prior_dic.values():
+                g.attrs["prior[{}]".format(prior.keywords['name'])] = (prior.keywords['prior_pars'][0],prior.keywords['prior_pars'][1],prior.func.__name__.encode('utf8'))
```
Comment (londumas) [inline py/picca/fitter2/chi2.py]: same as my comment, `prior.keywords['prior_pars'][0]` is hard to generalize for other type of priors.

**Comment by mblomqvist** [comment]
@londumas yes, exactly, the code is written for gaussian priors right now. It can be generalised if you really want.

**Comment by londumas** [comment]
@mblomqvist, my point is that by changing the way to read and write the parameters, we can easily generalize it in the future. Anyway, thank you for the work.

**Comment by mblomqvist** [comment]
@londumas OK, I'll try the generalisation

**Comment by londumas** [comment]
@mblomqvist, thank you very much

**Comment by londumas** [comment]
@mblomqvist, thank you.

**Comment by mblomqvist** [comment]
@londumas thanks for your input. The solution I came up with for saving is to convert the prior info into a list. Let me know if there is a smarter way.

**Comment by londumas** [comment]
@mblomqvist, it is perfect. Thank you very much

**Comment by mblomqvist** [comment]
ready to be merged

**Comment by ngbusca** [comment]
Thanks @mblomqvist !!

**Pull request merged**